tokio-fs/src/file.rs

```diff
@@ -0,0 +1,430 @@
+    /// file.read_to_end(&mut contents).await?;
+    ///
+    /// println!("len = {}", contents.len());
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn open<P>(path: P) -> io::Result<File>
+    where
+        P: AsRef<Path> + Send + Unpin + 'static,
+    {
+        let mut open_options = OpenOptions::new();
+        open_options.read(true);
+
+        open_options.open(path).await
+    }
+
+    /// Opens a file in write-only mode.
+    ///
+    /// This function will create a file if it does not exist, and will truncate
+    /// it if it does.
+    ///
+    /// See [`OpenOptions`] for more details.
+    ///
+    /// [`OpenOptions`]: struct.OpenOptions.html
+    ///
+    /// # Errors
+    ///
+    /// `CreateFuture` results in an error if called from outside of the Tokio
+    /// runtime or if the underlying [`create`] call results in an error.
+    ///
+    /// [`create`]: https://doc.rust-lang.org/std/fs/struct.File.html#method.create
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(async_await)]
+    ///
+    /// use tokio::fs::File;
+    /// use tokio::prelude::*;
+    ///
+    /// # async fn dox() -> std::io::Result<()> {
+    /// let mut file = File::create("foo.txt").await?;
+    /// file.write_all(b"hello, world!").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn create<P>(path: P) -> io::Result<File>
+    where
+        P: AsRef<Path> + Send + Unpin + 'static,
+    {
+        let std_file = asyncify(|| std::fs::File::create(&path)).await?;
+        Ok(File::from_std(std_file))
+    }
+
+    /// Convert a [`std::fs::File`][std] to a [`tokio_fs::File`][file].
+    ///
+    /// [std]: https://doc.rust-lang.org/std/fs/struct.File.html
+    /// [file]: struct.File.html
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// // This line could block. It is not recommended to do this on the Tokio
+    /// // runtime.
+    /// let std_file = std::fs::File::open("foo.txt").unwrap();
+    /// let file = tokio::fs::File::from_std(std_file);
+    /// ```
+    pub fn from_std(std: std::fs::File) -> File {
+        File { std }
+    }
+
+    /// Seek to an offset, in bytes, in a stream.
+    ///
+    /// Similar to `poll_seek`, but returning a `Future`.
+    ///
+    /// This method consumes the `File` and returns it back when the future
+    /// completes.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(async_await)]
+    ///
+    /// use tokio::fs::File;
+    /// use tokio::prelude::*;
+    ///
+    /// use std::io::SeekFrom;
+    ///
+    /// # async fn dox() -> std::io::Result<()> {
+    /// let mut file = File::open("foo.txt").await?;
+    /// file.seek(SeekFrom::Start(6)).await?;
+    ///
+    /// let mut contents = vec![0u8; 10];
+    /// file.read_exact(&mut contents).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[allow(clippy::needless_lifetimes)] // false positive: https://github.com/rust-lang/rust-clippy/issues/3988
```

Comment:
I dont see why we still need this

Reply:
The version of rustc / clippy we use has the bug still

Reply:
https://github.com/tokio-rs/tokio/runs/184842146

Reply:
ah must be a generator issue :/

Reply:
I believe it is fixed, but we would need to bump our toolchain.

Reply:
Yeah, #1384 is PR for that, but it blocked by a rustfmt bug.